features.html

```diff
@@ -1,322 +1,475 @@
+   <img class="gif" data-wait="true" data-scope="gif-scope" style="max-width: 480px" style="max-height: 160px"
+                          src="/assets/features/code_generation/rename_trait.png" data-gif="/assets/features/code_generation/rename_trait.gif"
+        alt="rename trait"
+   />
+    <!--p>You can rename symbols in-place or use a dialog with a preview.</p-->
   </div>
-
-  <div class="feature feature-major">
-    <h3>Intentions</h3>
-
-    <p>Intentions and quick fixes are micro refactorings automatically available depending on the context. If some
-      intentions are available, a light bulb icon appears in the editor. You can use <kbd>Alt+Enter</kbd> to invoke a
-      quick fix.</p>
-
-    <div class="feature-group grid">
-      <div class="feature">
-        {% include feature-gfy.html n="PessimisticFlashyAustraliancurlew" %}
-        <p><b>Contract module</b>: the opposite of <i>Expand Module</i> refactoring. Available only if there are no
-          child modules.</p>
-      </div>
-
-      <div class="feature">
-        {% include feature-gfy.html n="HauntingAgonizingDuck" %}
-        <p><b>Create module file</b>: if you have <code>mod foo;</code>, but no <code>foo.rs</code>
-          this quick fix will create the missing file.</p>
-      </div>
-
-      <div class="feature">
-        {% include feature-gfy.html n="BleakWholeGalapagospenguin" %}
-        <p><b>Extract inline module</b>: move <code>mod foo { ... }</code> to a separate file.</p>
-      </div>
-
-      <div class="feature" id="add-derive">
-        {% include feature-gfy.html n="ChubbyAcrobaticBrahmanbull" %}
-        <p><b>Add derive clause</b>: adds <code>#[derive(..)]</code> attribute to structs and enums.</p>
-      </div>
-
-      <div class="feature">
-        {% include feature-gfy.html n="CluelessGiddyBullfrog" %}
-        <p><b>Remove/Add curly braces</b>: toggles between <code>use foo::{bar}</code> and
-          <code>use foo::bar</code>.</p>
-      </div>
-
-      <div class="feature">
-        {% include feature-gfy.html n="ImperfectFreeGecko" %}
-        <p><b>Add missing fields</b>: a handy way to fill out the struct literal template.</p>
-      </div>
-
-      <div class="feature">
-        {% include feature-gfy.html n="GlisteningCostlyAntarcticfurseal" %}
-        <p><b>Implement methods</b>: quickly scafold an impl.</p>
-      </div>
-    </div>
-
+  <div class="item_bottomtext3">
+    <p>At any place in your code, call <strong>Rector This</strong> (<code>Ctrl+Shift+Alt+T / ⌃T</code>) to view the list of all context-depend refactorings. As an example, here is how you can easily put the result of an expression into a variable, affecting all usages:
```

Comment:
Maybe move part about `Refactor This` action into description and convert this block into `Extract Variable` section?